### ISSUE TITLE
CMake error "Cannot specify link libraries for target highfive_deps"

### ISSUE THREAD

**Issue by parsa** [BlueBrain/HighFive#271]
Hello,

After the recent commit (2f07be8153b1ad84cee9279958c98c8c67180c10) we now get a new CMake error while we configure our project (STEllAR-GROUP/phylanx#1090):
```
-- HDF5: Using hdf5 compiler wrapper to determine C configuration
-- Found HDF5: /usr/lib/x86_64-linux-gnu/hdf5/serial/libhdf5.so;/usr/lib/x86_64-linux-gnu/libpthread.so;/usr/lib/x86_64-linux-gnu/libsz.so;/usr/lib/x86_64-linux-gnu/libz.so;/usr/lib/x86_64-linux-gnu/libdl.so;/usr/lib/x86_64-linux-gnu/libm.so (found version "1.10.0.1")
-- HDF5 library version:  1.10.0.1
CMake Error at /usr/local/share/HighFive/CMake/HighFiveTargetDeps.cmake:31 (target_include_directories):
  Cannot specify include directories for imported target "highfive_deps".
Call Stack (most recent call first):
  /usr/local/share/HighFive/CMake/HighFiveConfig.cmake:24 (include)
  cmake/Phylanx_HighFive.cmake:34 (find_package)
  CMakeLists.txt:152 (phylanx_setup_highfive)


CMake Error at /usr/local/share/HighFive/CMake/HighFiveTargetDeps.cmake:32 (target_link_libraries):
  Cannot specify link libraries for target "highfive_deps" which is not built
  by this project.
Call Stack (most recent call first):
  /usr/local/share/HighFive/CMake/HighFiveConfig.cmake:24 (include)
  cmake/Phylanx_HighFive.cmake:34 (find_package)
  CMakeLists.txt:152 (phylanx_setup_highfive)
```

We are unclear on how to approach this error and wanted to ask for your advice with it.

This where and how we set up HDF5 and HighFive in our code:
https://github.com/STEllAR-GROUP/phylanx/blob/master/cmake/Phylanx_HighFive.cmake#L13-L34

and where we link against the HighFive target:
https://github.com/STEllAR-GROUP/phylanx/blob/2a061d60337b17322a805268a7e2b01677cf838c/src/plugins/fileio/CMakeLists.txt#L38-L41

Regards,
Phylanx Team

**Comment by tdegeus** [comment]
Hi @parsa ! Can you try if https://github.com/BlueBrain/HighFive/pull/272 fixes your issues?

**Comment by tdegeus** [comment]
No luck just yet @parsa , will let you know when it works again.

**Comment by parsa** [comment]
The issue goes away using CMake 3.11+ with both master and the fix in #272 but neither work with 3.10 and below. Since we raised our minimum CMake version from 3.3 to 3.11 yesterday we do not experience this issue anymore, but do you plan to roll out workaround for older CMakes or do you want to raise the minimum to 3.11?

**Comment by tdegeus** [comment]
I'm surprised that #272 does not work. I tried a minimal example with CMake 3.5 here: https://github.com/tdegeus/test_cmake . Can you elaborate what problem remains with #272 ?

**Comment by ferdonline** [comment]
Thanks for your report. We indeed want to keep compat with previous CMake, so this is a valid issue. We opened the more general #283

**Comment by ferdonline** [comment]
Fixed by #304
